unstructured/ingest/connector/pinecone.py

```diff
@@ -0,0 +1,143 @@
+import json
+import multiprocessing as mp
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, WriteError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    IngestDocSessionHandleMixin,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
```

Comment:
For things that need to be typed but don't want to include the import here, can we leverage the typing `TYPE_CHECKING` value:
```python
if t.TYPE_CHECKING:
    from pinecone import Index
```

This will help with type checks performed by mypy or an IDE.

Reply:
Thank you, this is helpful. Learned a bunch on type checks and imports as well while reading on this.

Adressed with [better type checking for session handles](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/0ec7cae1489f44ba5d2fde68596993b84bc4d380)